test/unit/services/test_system.py

```diff
@@ -2,284 +2,374 @@
-    "importance": "high",
-    "description": "Test description",
-    "parent_id": None,
-}
-
-SYSTEM_B_INFO_FULL = {
-    **SYSTEM_A_INFO,
-    "code": "test-name-b",
-    "created_time": MODEL_MIXINS_FIXED_DATETIME_NOW,
-    "modified_time": MODEL_MIXINS_FIXED_DATETIME_NOW,
-}
-
-
-def test_create(
-    test_helpers,
-    system_repository_mock,
-    model_mixins_datetime_now_mock,  # pylint: disable=unused-argument
-    system_service,
-):
-    """
-    Test creating a System
-
-    Verify that the `create` method properly handles the System to be created, generates the code,
-    and calls the repository's create method
-    """
-    system = SystemOut(id=str(ObjectId()), **SYSTEM_A_INFO_FULL)
-
-    # Mock `create` to return the created System
-    test_helpers.mock_create(system_repository_mock, system)
-
-    created_system = system_service.create(SystemPostSchema(**SYSTEM_A_INFO))
-
-    system_repository_mock.create.assert_called_with(SystemIn(**SYSTEM_A_INFO_FULL))
-    assert created_system == system
-
-
-def test_create_with_parent_id(
-    test_helpers,
-    system_repository_mock,
-    model_mixins_datetime_now_mock,  # pylint: disable=unused-argument
-    system_service,
-):
-    """
-    Test creating a System with a parent ID
-
-    Verify that the `create` method properly handles the System to be created when it has a parent ID
-    """
-    system_info = {
-        **SYSTEM_B_INFO,
-        "parent_id": str(ObjectId()),
-    }
-    full_system_info = {
-        **system_info,
-        "code": SYSTEM_B_INFO_FULL["code"],
-        "created_time": SYSTEM_B_INFO_FULL["created_time"],
-        "modified_time": SYSTEM_B_INFO_FULL["modified_time"],
-    }
-    system = SystemOut(id=str(ObjectId()), **full_system_info)
-
-    # Mock `get` to return the parent system
-    test_helpers.mock_get(
+
+class SystemServiceDSL:
+    """Base class for SystemService unit tests"""
+
+    test_helpers: ServiceTestHelpers
+    wrapped_utils: Mock
+    mock_system_repository: Mock
+    system_service: SystemService
+
+    @pytest.fixture(autouse=True)
+    def setup(
+        self,
+        test_helpers,
         system_repository_mock,
-        SystemOut(id=system.parent_id, **SYSTEM_B_INFO_FULL),
-    )
+        system_service,
+        # Ensures all created and modified times are mocked throughout
+        # pylint: disable=unused-argument
+        model_mixins_datetime_now_mock,
+    ):
+        """Setup fixtures"""
 
-    # Mock `create` to return the created System
-    test_helpers.mock_create(system_repository_mock, system)
+        self.test_helpers = test_helpers
+        self.mock_system_repository = system_repository_mock
+        self.system_service = system_service
 
-    created_system = system_service.create(SystemPostSchema(**system_info))
+        with patch("inventory_management_system_api.services.system.utils", wraps=utils) as wrapped_utils:
+            self.wrapped_utils = wrapped_utils
+            yield
```

Comment:
Not sure I know what the purpose of the wrapper here is but lets have a chat.

Reply:
I should put a comment here.